runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -130,8 +129,8 @@ public void channelRead(ChannelHandlerContext ctx, Object m)
         log.info("channelRead: Handshake succeeded. Server Corfu Version: [{}]", corfuVersion);
         log.debug("channelRead: There are [{}] messages in queue to be flushed.", this.messages.size());
         // Flush messages in queue
-        for (CorfuMsg message : this.messages) {
-            ctx.writeAndFlush(message);
+        while (!messages.isEmpty()) {
+            ctx.writeAndFlush(messages.poll());
```

Comment:
The messages are just being iterated, don't you actually want to drain the queue?

Reply:
Actually the queue is being drained and hence the poll(). Can you elaborate on why this is just an iteration?

Reply:
Hmm, this was suppose to be a comment on the diff:

for (CorfuMsg message : this.messages) {
            ctx.writeAndFlush(message);
}

Is this leaking actually a leak on master?

Reply:
I see. correct, master still is not draining.